scripts/check_types.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+#
+# Run mypy on new files changed or added in git.
+#
+# Usage: scripts/check_types.sh <python-path> <python-path>
+#
+# <python-path> is the path to run with mypy. If not specified, mypy with run
+# off of the files found in git status. You can run multiple paths with mypy
+set -e
+
+cd "$(dirname "$0")/.."
+
+PATHS=${*:-$(git status -s | sed "s@^[^ ]* @$PWD/@")}
```

Comment:
@derekadombek so I just tested this script out with the following input

```sh
./scripts/check_types.sh src/
```

My hope was that it would type check all the files in the src directory, however I just got this message "Either no git changes or no python files"

Reply:
Oh thats a good catch I dont think tested that use case. I'll try and add that as well.

Reply:
updated a bunch.. give it a try or update it how youd like. I got it close but there are a lot of weird `mypy` errors I just really dont understand tbh.

Reply:
I just tried the following and got an error.

```sh
❯ ./scripts/check_types.sh src/recordlinker/models
from full dir
mypy: can't decode file 'src/recordlinker/models/__pycache__/__init__.cpython-312.pyc': 'utf-8' codec can't decode byte 0xcb in position 0: invalid continuation byte
```

Reply:
I was thinking about something more like this

```diff
diff --git a/scripts/check_types.sh b/scripts/check_types.sh
old mode 100644
new mode 100755
index 7da683a..bba8f6f
--- a/scripts/check_types.sh
+++ b/scripts/check_types.sh
@@ -10,48 +10,9 @@ set -e
 
 cd "$(dirname "$0")/.."
 
-PATHS=${*:-$(git status -s | sed "s@^[^ ]* @$PWD/@")}
-
-# non_python_list=()
-# for p in ${PATHS}; do
-#   echo "$p"
-#   if [[ $p == *.py ]]; then
-#       mypy $p --ignore-missing-imports
-#   else
-#       non_python_list+=(1)
-#   fi
-# done
-
-full_directory() {
-  for p in ${PATHS}**/*/; do
-    target="src/recordlinker/linkage/"
-    if [ "$p" != "$target" ]; then
-      for f in $p**; do
-        mypy "$f" --ignore-missing-imports --follow-imports=skip
-      done
-    fi
-  done
-}
-
-single_files() {
-  non_python_list=()
-  for p in ${PATHS}; do
-    if [[ $p == *.py ]]; then
-        echo "$p"
-        mypy $p --ignore-missing-imports --follow-imports=skip
-    fi
-  done
-
-  if [[ -n ${non_python_list[@]} ]]; then
-      echo "Either no git changes or no python files"
-  fi
-}
-
-if [[ -d $PATHS  ]]; then
-    echo "from full dir"
-    full_directory
-else
-    echo "from single dir"
-    single_files
+PATHS=("$@")
+if [ -z "$PATHS" ]; then
+  PATHS=("src/recordlinker/linking" "src/recordlinker/models")
 fi
 
+mypy --follow-imports=silent ${PATHS[@]}
```